include/pcmki/pcmki_scheduler.h

```diff
@@ -107,4 +107,30 @@ extern gboolean show_scores;
 extern gboolean show_utilization;
 extern const char *transition_idle_timeout;
 
+/*!
+ * \internal
+ * \brief Check whether colocation's left-hand preferences should be considered
+ *
+ * \param[in] colocation  Colocation constraint
+ * \param[in] rsc         Right-hand instance (normally this will be
+ *                        colocation->rsc_rh, which NULL will be treated as,
+ *                        but for clones or bundles with multiple instances
+ *                        this can be a particular instance)
+ *
+ * \return true if colocation influence should be effective, otherwise false
+ */
+static inline bool
+pcmk__colocation_has_influence(const pcmk__colocation_t *colocation,
+                               const pe_resource_t *rsc)
+{
+    if (rsc == NULL) {
+        rsc = colocation->rsc_rh;
```

Comment:
There's no way that colocation->rsc_rh could be NULL, is there?

Reply:
No, colocations can only be created by pcmk__new_colocation(), which won't create one without it